android/java/org/chromium/base/BravePreferenceKeys.java

```diff
@@ -56,4 +59,49 @@ public final class BravePreferenceKeys {
     public static final String BRAVE_DEFERRED_DEEPLINK_PLAYLIST =
             "brave_deferred_deeplink_playlist";
     public static final String BRAVE_DEFERRED_DEEPLINK_VPN = "brave_deferred_deeplink_vpn";
+    public static final String BRAVE_CLOSE_TABS_ON_EXIT = "close_tabs_on_exit";
+    public static final String BRAVE_CLEAR_ON_EXIT = "clear_on_exit";
+    public static final String BRAVE_QUICK_ACTION_SEARCH_AND_BOOKMARK_WIDGET_TILES =
+            "org.chromium.chrome.browser.widget.quickactionsearchandbookmark.QuickActionSearchAndBookmarkWidgetProvider.TILES";
+
+    // These are dynamic keys
+    public static final String BRAVE_RECYCLERVIEW_POSITION = "recyclerview_visible_position_";
+    public static final String BRAVE_RECYCLERVIEW_OFFSET_POSITION = "recyclerview_offset_position_";
+
+    /*
+     * Returns the list of Brave's preference keys.
+     */
+    private static List<String> getKeysInUse() {
```

Comment:
This doesn't seem right to me, but https://github.com/brave/chromium/commit/b244bd1 is a broken link so I can't see what changed in upstream. Please use chromium.googlesource links for upstream changes

Reply:
Here is the change https://source.chromium.org/chromium/chromium/src/+/b244bd1dfd53472a71d38be65ffef915f15a0af4
Here we basically list our keys to avoid exception in `StrictPreferenceKeyChecker#checkIsKeyInUse`

Reply:
yes, but I think this is a symptom of another problem because this list should be auto generated. Adding them here manually does not seem like the right fix.

Reply:
well, nvm, it seems that this is exactly what upstream does https://source.chromium.org/chromium/chromium/src/+/main:chrome/browser/preferences/android/java/src/org/chromium/chrome/browser/preferences/ChromePreferenceKeys.java;l=50;drc=977dc02c431b4979e34c7792bc3d646f649dacb4;bpv=1;bpt=1